solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -252,14 +251,14 @@ def run_persistence(session, observation, forecast, run_time, issue_time,
     forecast_start, forecast_end = utils.get_forecast_start_end(
         forecast, issue_time, False)
     intraday = utils._is_intraday(forecast)
-    if not intraday:
-        # raise ValueError if not intraday and not midnight to midnight
-        utils._check_midnight_to_midnight(forecast_start, forecast_end)
 
     if load_data is None:
         load_data = _default_load_data(session)
     data_start, data_end = utils.get_data_start_end(
-        observation, forecast, run_time)
+        observation, forecast, run_time, issue_time)
+    if data_end > run_time:
+        raise ValueError(
+            'Persistence forecast requires data from after run_time')
```

Comment:
They don't at the moment and I'm not sure they should. I should add documentation for this but I think at least day ahead and week ahead should not rely on run time to set data start/end. Basically, you are able to determine what data went into the persistence forecast without worrying about when it was made. Therefore, it's possible that the run time is not after data end, so fail instead of not having enough data for the forecast.

Reply:
that makes sense and now I agree